sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +697,69 @@ const StartHTLCAssetLockListener = (
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+// Byzantine Swaps: Timed counterpart for timed AssetLockListener for reversion
+const assetLockExpirationCallback = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    // Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+
+    const reclaimCallback = async (contract: Contract, assetType: string, assetID: string, recipientECert: string, contractID: string) => {
+        // Check if asset hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract, contractID));
+        
+        if (islocked == false) {
+            // CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            // CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                // If it succeeds, no change
+                const [reclaimResult, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                if (isReclaimableQueryError) {
+                    reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);             
+                }
+
+                // Put retry logic in event of failure. Retry 3x and then give up if unsuccessful (temp solution for now; requires CORE changes)
+                // If it fails, retry for i (arbitrary) more times until you quit
+                let i = 0;
+                do {
+                    let [retryReclaimResult, retryReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                    if (!retryReclaimableQueryError) {
+                        break;
+                    }
+
+                    i++;
+                } while (i < 3);
+            }
+        }
+    }
+
+    // If you have time remaining, call setTimeout with reclaimCallback
+    if (expiryTime - currTimeSecs > 0) {
+        setTimeout(reclaimCallback, 1000 * (expiryTime - currTimeSecs), contract, assetType, assetID, recipientECert, hash);
```

Comment:
the parameters (from 3rd parameter and on, after the time) that you are passing here is wrong, it should be same as in 759 `contract, assetType, assetID, recipientECert, contractID`.
Also I think now you can remove `assetType, assetID, recipientECert` these parameters from ReclaimCallback, and keep only `contract, contractID`.